dubbo-common/src/main/java/org/apache/dubbo/common/config/Environment.java

```diff
@@ -212,11 +257,17 @@ public void setDynamicConfiguration(DynamicConfiguration dynamicConfiguration) {
 
     @Override
     public void destroy() throws IllegalStateException {
-        clearExternalConfigs();
-        clearAppExternalConfigs();
+        externalConfiguration = null;
+        appExternalConfiguration = null;
+        appConfiguration = null;
         globalConfiguration = null;
         dynamicConfiguration = null;
         dynamicGlobalConfiguration = null;
+        initConfigs();
```

Comment:
In unit testing, it is sometimes necessary to start DubboBootstrap twice, and the `Environment` may need to be reused after destroy.

Reply:
We can call `initConfigs()` directly, if `Environment` need to be resued after destory. If we make all of them tegother, the method `destory` makes no sense.

Unit testing is the same as that.

Reply:
You're right, I try move `initConfigs()` into `initialize()`. It will be initialized again if reset dubbo:  
DubboBootstrap#initialize() -> ApplicationModel#initFrameworkExts() -> Environment#initialize()

Reply:
There is still a problem, the `DubboBootstrap.reset()` may be called after initialize environment and before test, so I had to add a `reset` method to `Environment`.
DubboBootstrap.reset() -> ApplicationModel.reset() -> Environment.reset()